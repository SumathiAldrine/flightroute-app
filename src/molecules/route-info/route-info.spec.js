import React from 'react';
import renderer from 'react-test-renderer';
import RouteInformation from './index';

describe('Test Route Information', () => {
  it('renders Route Information for desktop', () => {
  const mockHandleSelect = jest.fn();
  const props = {
    selected: true,
    handleSelect: mockHandleSelect, 
    index: 1, 
    item: {
      id :1, 
      flightNo :'EK500', 
      departureDate:'12Dec2021', 
      departureTime:'08:30', 
      arrivalDate:'12Dec2021', 
      arrivalTime:'11:30', 
      duration:'3hrs 14mins', 
      origin: 'DXB', 
      destination: 'BOM', 
      aircraft: 'B777'
    }
  };
  const tree = renderer.create(<RouteInformation isMobile ={false} {...props}/>).toJSON();
  expect(tree).toMatchSnapshot();
  });

  it('renders Route Information for mobile', () => {
    const mockHandleSelect = jest.fn();
    const props = {
      selected: false,
      handleSelect: mockHandleSelect, 
      index: 1, 
      item: {
        id :1, 
        flightNo :'EK500', 
        departureDate:'12Dec2021', 
        departureTime:'08:30', 
        arrivalDate:'12Dec2021', 
        arrivalTime:'11:30', 
        duration:'3hrs 14mins', 
        origin: 'DXB', 
        destination: 'BOM', 
        aircraft: 'B777'
      }
    };
    const tree = renderer.create(<RouteInformation isMobile ={true} {...props}/>).toJSON();
    expect(tree).toMatchSnapshot();
    });

});