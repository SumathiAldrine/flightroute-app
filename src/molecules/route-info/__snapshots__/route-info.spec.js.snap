// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Route Information renders Route Information for desktop 1`] = `
<li
  className="route-info__container"
  id={1}
  tabIndex="0"
>
  <div
    className="leg-information__leg-info"
  >
    <div
      className="leg-information__leg-info-name"
      tabIndex="0"
    >
      DXB
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      08:30
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      12Dec2021
    </div>
  </div>
  <div
    className="aircraft-info__aircraft"
  >
    <div
      className="aircraft-info__tail-container"
    >
      <img
        alt="Tail"
        className="aircraft-info__flight-tail"
        src="tail_icon.png"
      />
      <span
        className="aircraft-info__tailNo"
        tabIndex="0"
      >
        B777
      </span>
    </div>
    <div
      className="aircraft-info__progress-container"
    >
      <div
        className="aircraft-info__progress-line"
      >
        <img
          alt="Airline"
          className="aircraft-info__progress-airplane"
          src="flight_icon.png"
        />
        <div
          className="aircraft-info__progress-line--complete"
        />
        <div
          className="aircraft-info__progress-point aircraft-info__progress-point--finish"
        />
      </div>
    </div>
    <div
      className="aircraft-info__duration"
      tabIndex="0"
    >
      3hrs 14mins
    </div>
  </div>
  <div
    className="leg-information__leg-info"
  >
    <div
      className="leg-information__leg-info-name"
      tabIndex="0"
    >
      BOM
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      11:30
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      12Dec2021
    </div>
  </div>
  <div
    className="flight-action__select"
  >
    <div
      className="flightNo"
      tabIndex="0"
    >
      EK500
    </div>
    <button
      className="cta"
      onClick={[MockFunction]}
    >
      Select
    </button>
  </div>
</li>
`;

exports[`Test Route Information renders Route Information for mobile 1`] = `
<li
  className="route-info__container"
  id={1}
  tabIndex="0"
>
  <div
    className="leg-information__leg-info"
  >
    <div
      className="leg-information__leg-info-name"
      tabIndex="0"
    >
      DXB
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      08:30
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      12Dec2021
    </div>
  </div>
  <div
    className="aircraft-info__aircraft"
  >
    <div
      className="aircraft-info__tail-container"
    >
      <img
        alt="Tail"
        className="aircraft-info__flight-tail"
        src="tail_icon.png"
      />
      <span
        className="aircraft-info__tailNo"
        tabIndex="0"
      >
        B777
      </span>
    </div>
    <div
      className="aircraft-info__progress-container"
    >
      <div
        className="aircraft-info__progress-line"
      >
        <img
          alt="Airline"
          className="aircraft-info__progress-airplane"
          src="flight_icon.png"
        />
        <div
          className="aircraft-info__progress-line--complete"
        />
        <div
          className="aircraft-info__progress-point aircraft-info__progress-point--finish"
        />
      </div>
    </div>
    <div
      className="aircraft-info__duration"
      tabIndex="0"
    >
      3hrs 14mins
    </div>
  </div>
  <div
    className="leg-information__leg-info"
  >
    <div
      className="leg-information__leg-info-name"
      tabIndex="0"
    >
      BOM
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      11:30
    </div>
    <div
      className="leg-information__leg-info-time"
      tabIndex="0"
    >
      12Dec2021
    </div>
  </div>
  <div
    className="flight-action__select"
  >
    <div
      className="flightNo"
      tabIndex="0"
    >
      EK500
    </div>
    <button
      className="cta"
      onClick={[MockFunction]}
    >
      Select
    </button>
  </div>
</li>
`;
